test/test_testing.py

```diff
@@ -2681,6 +2677,20 @@ def test_container_pebble_ready(self):
             ]
         )
 
+    def test_get_filesystem_root(self):
+        harness = ops.testing.Harness(ops.CharmBase, meta='''
+            name: test-app
+            containers:
+              foo:
+                resource: foo-image
+        ''')
+        foo_root = harness.get_filesystem_root("foo")
+        self.assertTrue(foo_root.exists())
+        self.assertTrue(foo_root.is_dir())
+        harness.begin()
+        container = harness.charm.unit.get_container("foo")
+        self.assertEqual(foo_root, harness.get_filesystem_root(container))
```

Comment:
This is a good test as far as it goes, but I think we need a few more test cases to test that 1) files that the test writes via `get_filesystem_root` can then be pulled and listed, and 2) files pushed and make_dir'd can then be accessed via `get_filesystem_root`. Or am I missing something?

Reply:
Yes, we already have a suite of tests for the filesystem, using the Pebble APIs (blackbox tests). I think I will add some more tests related to the `get_filesystem_root` method, since that's an exposed API now. Thanks!

Reply:
I have added more unit tests for the `get_filesystem_root` method, thanks!